alphadia/planning.py

```diff
@@ -329,6 +329,7 @@ def run(
                 workflow.timing_manager.set_start_time("extraction")
                 psm_df, frag_df = workflow.extraction()
                 workflow.timing_manager.set_end_time("extraction")
+                workflow.timing_manager.save()
```

Comment:
I have moved the saving of the timing manager to this point (rather than saving it with each call) because it helps with error handling. At the moment we check if the `timing_manager.pkl` file exists and fail gracefully (by moving to the next file instead of crashing) if it does not. However, the search will crash if the ``timing_manager.pkl`` file exists but is incomplete because an error arose after it was first saved but before it was complete. This should stop such errors arising in the future.